data/functions.sql

```diff
@@ -585,6 +601,38 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql STABLE;
 
+CREATE OR REPLACE FUNCTION mz_cycling_network_(way_tags hstore, osm_id bigint)
+RETURNS text AS $$
+DECLARE
+  networks text[] := ARRAY(
+    SELECT hstore(tags)->'network'
+    FROM planet_osm_rels
+    WHERE parts && ARRAY[osm_id]
+      AND parts[way_off+1:rel_off] && ARRAY[osm_id]
+      AND mz_is_path_major_route_relation(hstore(tags)));
+BEGIN
+  RETURN CASE
+    WHEN networks && ARRAY['icn'] THEN 'iwn'
+    WHEN networks && ARRAY['ncn'] THEN 'ncn'
+    WHEN way_tags->'ncn'='yes' OR way_tags ? 'ncn_ref' THEN 'ncn'
+    WHEN networks && ARRAY['rcn'] THEN 'rcn'
+    WHEN way_tags->'rcn'='yes' OR way_tags ? 'rcn_ref' THEN 'rcn'
+    WHEN networks && ARRAY['lcn'] THEN 'lcn'
+    WHEN way_tags->'lcn'='yes' OR way_tags ? 'lcn_ref' THEN 'lcn'
+  END;
+END;
+$$ LANGUAGE plpgsql STABLE;
+
+CREATE OR REPLACE FUNCTION mz_cycling_network(way_tags hstore, osm_id bigint)
+RETURNS text AS $$
+BEGIN
+  RETURN CASE
+    WHEN way_tags->'icn' = 'yes' OR way_tags ? 'icn_ref' THEN 'icn'
+    ELSE mz_cycling_network_(way_tags, osm_id)
```

Comment:
This was my attempt to try and avoid the relatively expensive relation query if the way tags were already `icn`. I couldn't figure out a way to inline it all into the case branch.

Turns out that I wasn't using `mz_calculate_is_bicycle_route` anywhere. I removed it in 72d48a1.